api/types/fix.go

```diff
@@ -9,14 +9,15 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-// FixKustomizationPreUnmarshalling modies the raw data
+// FixKustomizationPreUnmarshalling modifies the raw data
 // before marshalling - e.g. changes old field names to
 // new field names.
 func FixKustomizationPreUnmarshalling(data []byte) ([]byte, error) {
-	deprecateFieldsMap := map[string]string{
-		"imageTags:": "images:",
+	deprecatedFieldsMap := map[string]string{
+		"imageTags:":      "images:",
+		"patchesJson6902": "patches",
```

Comment:
What if a kustomization file contains both `patchesJson6902` and `patches` field?

Reply:
good question :)
what happens now?

In a `build` context, the simplest thing would be to try to apply both, in whatever order they appeared.  Or even simpler, just look for that immediately, and error out, so we don't have to do a more sophisticated check of patch conflicts.

The `fix` command aspires to fix errors and output a valid, up-to-date kustomization.yaml file.

So `fix` should parse in all patches, as much as possible make sure they are orthogonal (i suspect its rare that people would have both - but we need the crawler to know for sure), then write them out to a kustomization file that has only a `patches` directive.